rusk-profile/src/lib.rs

```diff
@@ -207,45 +107,111 @@ pub fn verify_common_reference_string(buff: &[u8]) -> bool {
     hash == CRS_17
 }
 
-pub fn get_lockfile_path() -> PathBuf {
-    let mut dir = env::current_dir().expect("Failed to get current dir");
-    let mut lockfile = dir.join("Cargo.lock");
+pub fn clean_outdated_keys(ids: &Vec<[u8; 32]>) -> Result<(), io::Error> {
+    info!("Cleaning outdated keys (if any)");
+    let ids_as_str: Vec<String> =
+        ids.iter().map(|id| hex::encode(id)).collect();
+
+    let entries_to_delete = fs::read_dir(&get_rusk_keys_dir()?)?
+        .map(|res| res.map(|e| e.path()))
+        .filter(|res| res.is_ok())
+        .map(|res| res.unwrap())
+        .filter(|e| e.is_file())
+        .filter(|p| match p.extension() {
+            Some(os_str) => match os_str.to_str() {
+                Some("pk" | "vd") => match p.file_stem() {
+                    Some(os_str) => match os_str.to_str() {
+                        Some(id) => !ids_as_str.contains(&id.to_string()),
+                        _ => true,
+                    },
+                    _ => true,
+                },
+
+                _ => true,
+            },
+            None => true,
+        })
+        .collect::<Vec<_>>();
 
-    while !lockfile.exists() {
-        if !dir.pop() {
-            panic!("Failed to fetch Cargo.lock from upper dir!");
-        }
+    for entry in entries_to_delete {
+        let ext = entry.extension();
+        let id = entry.file_stem();
+
+        // We can safely unwrap here since in the inter above we already checked
+        // that this is always Some(&str)
+        let mut path = PathBuf::from(id.unwrap());
+        path.set_extension(ext.unwrap());
```

Comment:
Actually, you can't. You could still have file with no extension (or with no name but only extension) that are going to be deleted.